fix http message parser and javascript lib for rdsn calls

imzhenyu opened the pull request:
fix http message parser due to:
- OPTIONAL is used for http request when cross-domain http msg is used
- custom headers are forbidden when cross-domain http msgs are used
- http msgs are splitted or batched in one TCP read (due to TCP streaming) (cannot assume each read only returns one whole msg)

fix unnecessary "parse failed" due to uninitialized read_next in parsers::get_message_on_receive

imzhenyu commented:
This is to fix #420 , now cli works, but other components in web studio are still broken, to be fixed later.

qinzuoyan commented on src/core/tools/common/http_message_parser.cpp:
```diff
@@ -57,47 +55,60 @@ char(&ArraySizeHelper(T(&array)[N]))[N];
+        
+        // hash
+        char *end;
+        hdr->client.hash = std::strtoull(args[1].c_str(), &end, 10);
```
do not check the end

imzhenyu replied:
done.

Commit:
fix http message parser due to:

    - OPTIONAL is used for http request when cross-domain http msg is used
    - custom headers are forbidden when cross-domain http msgs are used
    - http msgs are splitted or batched in one TCP read (due to TCP streaming) (cannot assume each read only returns one

    fix unnecessary "parse failed" due to uninitialized read_next in parsers::get_message_on_receive

Commit:
fix JavaScript code generation and libraries due to http message parser change

qinzuoyan commented:
looks good to me
